packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ServiceClientBuilderTemplate.java

```diff
@@ -465,11 +471,33 @@ protected void writeSyncClientBuildMethod(AsyncSyncClient syncClient, AsyncSyncC
 
     protected void writeSyncClientBuildMethodFromInnerClient(AsyncSyncClient syncClient, JavaBlock function,
         String buildMethodName, boolean wrapServiceClient) {
+
+        boolean writeInstrumentation = !JavaSettings.getInstance().isAzureV1();
+
+        if (writeInstrumentation) {
+            function.line(
+                "HttpInstrumentationOptions localHttpInstrumentationOptions = this.httpInstrumentationOptions == null ? new HttpInstrumentationOptions() : this.httpInstrumentationOptions;");
+
+            boolean hasEndpoint = syncClient.getClientBuilder()
+                .getBuilderTraits()
+                .stream()
+                .anyMatch(trait -> trait instanceof EndpointTrait);
+
+            function.line(
+                "SdkInstrumentationOptions sdkInstrumentationOptions = new SdkInstrumentationOptions(%1$s).setSdkVersion(%2$s)%3$s;",
+                "PROPERTIES.getOrDefault(SDK_NAME, \"UnknownName\")", "PROPERTIES.get(SDK_VERSION)",
+                hasEndpoint ? ".setEndpoint(this.endpoint)" : "");
```

Comment:
Looking at `buildInnerClient` for `ApiKeyClientBuilder`, if the `endpoint` has a default value should that same logic be used here?

Reply:
yes, another great catch!